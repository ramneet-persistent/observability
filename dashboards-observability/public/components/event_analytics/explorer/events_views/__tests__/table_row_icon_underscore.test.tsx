/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { DocViewTableRowIconUnderscore } from '../detail_table/table_row_icon_underscore';

describe('DocViewTableRowIconUnderscore component', () => {
  configure({ adapter: new Adapter() });

  it('Renders DocViewTableRowIconUnderscore component', async () => {
    const wrapper = mount(<DocViewTableRowIconUnderscore />);

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
