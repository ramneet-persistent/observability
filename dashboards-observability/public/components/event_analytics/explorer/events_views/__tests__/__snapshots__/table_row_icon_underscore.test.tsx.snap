// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocViewTableRowIconUnderscore component Renders DocViewTableRowIconUnderscore component 1`] = `
<DocViewTableRowIconUnderscore>
  <EuiIconTip
    aria-label="Warning"
    color="warning"
    content="Field names beginning with _ are not supported"
    iconProps={
      Object {
        "className": "osdDocViewer__warning",
        "data-test-subj": "underscoreWarning",
      }
    }
    size="s"
    type="alert"
  >
    <EuiToolTip
      content="Field names beginning with _ are not supported"
      delay="regular"
      position="top"
    >
      <span
        className="euiToolTipAnchor"
        onKeyUp={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <EuiIcon
          aria-label="Warning"
          className="osdDocViewer__warning"
          color="warning"
          data-test-subj="underscoreWarning"
          onBlur={[Function]}
          onFocus={[Function]}
          size="s"
          tabIndex={0}
          type="alert"
        >
          <EuiIconEmpty
            aria-hidden={true}
            aria-label="Warning"
            className="euiIcon euiIcon--small euiIcon--warning euiIcon-isLoading osdDocViewer__warning"
            data-test-subj="underscoreWarning"
            focusable="true"
            onBlur={[Function]}
            onFocus={[Function]}
            role="img"
            style={null}
            tabIndex={0}
          >
            <svg
              aria-hidden={true}
              aria-label="Warning"
              className="euiIcon euiIcon--small euiIcon--warning euiIcon-isLoading osdDocViewer__warning"
              data-test-subj="underscoreWarning"
              focusable="true"
              height={16}
              onBlur={[Function]}
              onFocus={[Function]}
              role="img"
              style={null}
              tabIndex={0}
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            />
          </EuiIconEmpty>
        </EuiIcon>
      </span>
    </EuiToolTip>
  </EuiIconTip>
</DocViewTableRowIconUnderscore>
`;
